src/box/memtx_tree.cc

```diff
@@ -921,8 +921,8 @@ memtx_tree_index_get_internal(struct index *base, const char *key,
 		memtx_tree_find(&index->tree, &key_data);
 	if (res == NULL) {
 		*result = NULL;
-		if (part_count == cmp_def->part_count)
-			memtx_tx_track_point(txn, space, base, key);
+		assert(part_count == cmp_def->unique_part_count);
+		memtx_tx_track_point(txn, space, base, key);
```

Comment:
I don't get it. Doesn't `cmp_def` have the same value in `part_count` and `unique_part_count` fields?

Reply:
Please read this comment:
https://github.com/tarantool/tarantool/blob/8ee0e434610189f3e4bbca0264de60aa79f0696e/src/box/key_def.h#L191-L198

Please look at this snippet:
https://github.com/tarantool/tarantool/blob/8ee0e434610189f3e4bbca0264de60aa79f0696e/src/box/index_def.c#L100-L105

Reply:
As for me, if key_def is unique, it shouldn't be merged with pk_def, and cmp_def will have the same value in part_count and unique_part_count fields. I will ask the team about it.